platforms/software/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/results/serializable/SerializableTestResult.java

```diff
@@ -170,7 +172,7 @@ public static SerializableTestResult deserialize(Decoder decoder) throws IOExcep
             }
 
             ImmutableList<SerializableFailure> failures = deserializeFailures(decoder);
-            ImmutableList<SerializedMetadata> metadatas = deserializeMetadatas(decoder);
+            ImmutableList<TestMetadataEvent> metadatas = deserializeMetadatas(decoder);
```

Comment:
🤔 Looking at the code here, it seems almost like an oversight that `TestMetadataEvent`s are both an input and output to serialization.